src/foam/core/IDSupport.js

```diff
@@ -143,11 +143,11 @@ foam.CLASS({
           {
             name: 'toString',
             code: function() {
-              var arr = [];
+              var str = '';
               for ( var prop in this.instance_ ) {
-                arr.push( this[prop] ? this[prop].toString() : '' );
+                str += ( this[prop] ? this[prop].toString() : '' ) + '-';
```

Comment:
so this can produce 
'-'  if this[prop] is null/undefined?  when does this happen?